client/state/data-layer/wpcom/sites/rewind/downloads/index.js

```diff
@@ -66,22 +70,112 @@ const fromApi = data => ( {
 	downloadCount: +data.downloadCount,
 	validUntil: data.validUntil,
 	url: data.url,
+	error: data.error,
 } );
 
-export const updateProgress = ( { dispatch }, { siteId, downloadId }, data ) =>
-	dispatch( updateRewindBackupProgress( siteId, downloadId, data ) );
+/**
+ * When requesting the status of a backup creation, it could be done by a specific download Id or without it.
+ * In the second case a collection is returned so we get the first and latest backup creation object.
+ * If an error property was found, it will display an error card stating it.
+ * Otherwise the backup creation progress will be updated.
+ *
+ * @param {function} dispatch Method to trigger a state change.
+ * @param {number}   siteId   Id of the site for the one we're creating a backup.
+ * @param {object}   apiData  Data returned by a successful response.
+ */
+export const updateProgress = ( { dispatch }, { siteId }, apiData ) => {
+	if ( ! isEmpty( apiData[ 0 ] ) ) {
+		const data = fromApi( apiData[ 0 ] );
+		dispatch(
+			get( data, [ 'error' ], false )
+				? rewindBackupUpdateError( siteId, data.downloadId, data )
+				: updateRewindBackupProgress( siteId, data.downloadId, data )
+		);
+	}
+};
 
+/**
+ * If the backup creation progress request fails, an error notice will be shown.
+ *
+ * @param {function} dispatch Method to trigger a state change.
+ *
+ * @returns {function} The dispatched action.
+ */
 export const announceError = ( { dispatch } ) =>
 	dispatch(
 		errorNotice(
 			translate( "Hmm, we can't update the status of your backup. Please refresh this page." )
 		)
 	);
 
+/**
+ * Mark a specific downloadable backup record as dismissed.
+ * This has the effect that subsequent calls to /sites/%site_id%/rewind/downloads won't return the download.
+ *
+ * @param {function} dispatch Method to trigger a state change.
+ * @param {object}   action   Changeset to update state.
+ *
+ * @returns {function} The dispatched action.
+ */
+export const dismissBackup = ( { dispatch }, action ) =>
+	dispatch(
+		http(
+			{
+				method: 'POST',
+				apiNamespace: 'wpcom/v2',
+				path: `/sites/${ action.siteId }/rewind/downloads/${ action.downloadId }`,
+				body: {
+					dismissed: true,
+				},
+			},
+			action
+		)
+	);
+
+/**
+ * On successful dismiss, the card will be removed and we don't need to do anything further.
+ * If request succeeded but backup couldn't be dismissed, a notice will be shown.
+ *
+ * @param {function} dispatch Method to trigger a state change.
+ * @param {object}   action   Changeset to update state.
+ * @param {data}     data     Description of request result.
+ */
+export const backupSilentlyDismissed = ( { dispatch }, action, data ) => {
+	if ( ! data.dismissed ) {
```

Comment:
Note that if we create a schema here we can have Calypso automatically fail-over here and thus we won't have to write two separate handlers that do the same thing…

```js
const fromDismissApi = ( { dismissed } ) => {
  if ( ! data.dismissed ) {
    throw false
  }
}

dispatchRequest( dismissBackup, null, dismissFailed )
```